### PR TITLE
Github Actions Test

### LINKED ISSUE #12: Create a GitHub Actions pipeline
The pipeline should build the code and run the tests. I recommend separating these two tasks into two different pipeline. 
Also, make sure to configure that pull requests can't be merged if the pipeline fails.

### PR DESCRIPTION
**Pull request by adamv17**
Test the pull requests coming into main. closes #12

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    name: Build and Run Tests
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v1
+      - name: Build and Run Tests
+        run: # execute your tests generating test results
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v2
+        if: always() # always run even if the previous step fails
```
Comment (barel13) [inline .github/workflows/tests.yml:19]: I don't you can test the code if you can't build your code...

**Review by barel13 — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+jobs:
+  build:
+    name: Build and Run Tests
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v1
+      - name: Build and Run Tests
+        run: # execute your tests generating test results
```
Comment (barel13) [inline .github/workflows/tests.yml:16]: Am I wrong or this action does nothing? (there is no run for testing)

Comment (adamv17) [reply .github/workflows/tests.yml:16]: Oops you are right

Comment (adamv17) [reply .github/workflows/tests.yml:16]: What would I need to add?

**Pull request closed without merging**